waku/waku_lightpush/callbacks.nim

```diff
@@ -0,0 +1,63 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+ ../waku_core,
+ ../waku_relay,
+ ./common,
+ ./protocol,
+ ../waku_rln_relay,
+ ../waku_rln_relay/protocol_types,
+ ../common/ratelimit
+import
+ std/times,
+ libp2p/peerid,
+ stew/byteutils
+
+proc generateAndValidateRLNProof*(rlnPeer: Option[WakuRLNRelay], message: WakuMessage): Result[WakuMessage, string] =
+  # TODO: check and validate if the message already has RLN proof?
+  if rlnPeer.isNone():
+    return ok(message) # publishing message without RLN proof
+
+  # generate and append RLN proof
+  let
+    time = getTime().toUnix()
+    senderEpochTime = float64(time)
+  var msgWithProof = message
+  let appendProofRes = rlnPeer.get().appendRLNProof(msgWithProof, senderEpochTime)
+  if appendProofRes.isOk():
+    return ok(msgWithProof)
+  
+  return err("RLN proof generation failed: " & appendProofRes.error)
+
+proc getNilPushHandler*(): PushMessageHandler =
+  return proc(
+    peer: PeerId, pubsubTopic: string, message: WakuMessage
+  ): Future[WakuLightPushResult[void]] {.async.} =
+    return err("no waku relay found")
+
+proc getRelayPushHandler*(
```

Comment:
Is it necessary to have proc producing another proc? 
Isn't it possible to define the necessary callback proc here and just use it as handler there?